labeling.py

```diff
@@ -0,0 +1,143 @@
+# face detection and printing errors to index file
+
+import os
+import sys
+from shutil import copyfile
+
+import cv2
+
+
+def label_faces():
+    """
+    This function detects faces in images in a directory and then places the bounding boxes in a txt file in YOLO
+    format.
+    It also creates an index file with images where it could not detect a face or detected more than one face for later
+    manual labelling.
+    """
+    # sets the emotion
+    try:
+        emotion = sys.argv[1]
+    except IndexError:
+        emotion = 'happy'
+        print('Emotion set to {}'.format(emotion))
+
+    # sets the image folder
+    try:
+        img_folder = sys.argv[2]
+    except IndexError:
+        print('No image folder entered, exiting program')
+        sys.exit(1)
+    emotion_folder = os.path.join(img_folder, emotion)
+    # checks for an index and a log folder and creates it if it does not exist
+    index = open(os.path.join(emotion_folder, 'index.txt'), 'w+')
+    log = open(os.path.join(emotion_folder, 'log.txt'), 'w+')
+
+    # cycles through each file in the directory
+    for img_file in os.listdir(emotion_folder):
+        if img_file.endswith(('.png', '.jpg')):
+            try:  # the cascade file includes weights for open-cv to detect faces
+                cascade_file = os.path.join(img_folder, 'haarcascade_frontalface_default.xml')
+            except OSError:
+                print('No cascade file detected, please download and place in directory before running')
+                sys.exit(1)
+            face_cascade = cv2.CascadeClassifier(cascade_file)  # face detection idk how it works
+            img = cv2.imread(os.path.join(emotion_folder, img_file))
```

Comment:
check if img is None or not. If the path got no image, it won't give us the right data.